apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerRetryOptions.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using System;
+using static Google.Api.Gax.Grpc.RetrySettings;
+
+namespace Google.Cloud.Spanner.Data
+{
+    /// <summary>
+    /// Options to use with <see cref="SpannerRetryHelpers"/>.
+    /// TODO: Consider making public or entirely eliminating once we know 
+    /// what we can reuse from GAX retry helpers.
+    /// </summary>
+    internal sealed class SpannerRetryOptions
+    {
+        internal SpannerRetryOptions(
```

Comment:
Do we need all of this anyway? For example, do we ever really need to change the exception filter?

Reply:
For retrying aborted commints:

- We shouldn't ever change the exception filter.
- As per discussion with the Spanner team, the backoff config is not modifiable by client code either.

But, I was trying to write a more general use kind of retry helper in case there are more "high level" retries we needed/wanted to offer, something that in the future, if needed across libraries, we could consider moving to Gax making the appropiate changes.

That said, I have no problem hardwiring some or all of this, but then I'd also like to have the retry method be specifically for aborted transactions and not for anything else. It would be weird to have the general method use the hardwire config (for instance, for the filter) that only really makes sense for aborted transactions.

Or should we consider modifying Gax to extract general retry logic that can be reused on a per call retry and also on higher level retries?

Reply:
My experience is that there tends to be enough that's "just a bit different" in each application-level retry that it's not worth trying to encapsulate too much of it.
If there are aspects of GAX that can be easily exposed, I'm open to that.